Nc-News/src/components/Articles.jsx

```diff
@@ -1,28 +1,34 @@
 import { useEffect, useState } from "react";
 import { getArticles } from "../api";
 import { ArticleCard } from "./Containers/ArticleCard";
+import { useNavigate, useParams } from "react-router-dom";
 
 export function Articles() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const { slug } = useParams();
+  const params = { params: { topic: slug } };
 
   useEffect(() => {
     setIsLoading(true);
-    getArticles().then((response) => {
+    getArticles({ params: { topic: slug } }).then((response) => {
       setArticles(response.data.articles);
       setIsLoading(false);
     });
-  }, []);
+  }, [slug]);
```

Comment:
Brilliant that you've refactored this component into one that can also be used for the topics endpoint, rather than make a new component ⭐